tests/tests.js

```diff
@@ -0,0 +1,259 @@
+// Tests for checkit app
+
+// Tests for checkit.js
+
+// Testing Month Object
+QUnit.module( "Month Tests", {
+// Hardcoded a date for moment object because javascript's Date object
+// does not have a way to get the number of days in the month.
+
+
+beforeEach: function() {
+    // Prepare something once for all tests
+    this.valentine = moment("2017-02-14", "YYYY-MM-DD");
+    this.dateString = this.valentine.format("YYYYMMDD");
+    this.testmonth = new Month(this.dateString);
+}
+
+});
+
+QUnit.test("init month object test", function(assert) {
+    
+    assert.expect(13);
+    assert.equal(this.testmonth.dateString, this.dateString, "dateString property");
+    assert.ok(moment.isMoment(this.testmonth.date), "testmonth.date is a moment obj");
+    assert.equal(this.testmonth.firstActiveDayIndex, 2, "Initialized first active day index");
+    assert.equal(this.testmonth.numberOfDays, 28, "Initialized number of days");
+    assert.equal(this.testmonth.monthYear, 2017, "Initialized the year");
+    
+    assert.equal(this.testmonth.monthIndex, 1, "Initialized the index of the month");
+    assert.equal(this.testmonth.monthName, "February", "Initialized the month's name");
+    assert.equal(this.testmonth.startDay, 14, "Initialized the start day");
+    
+    assert.ok(moment.isMoment(this.testmonth.firstDayDate), "testmonth.firstDayDate is a moment obj");
+    assert.equal(this.testmonth.firstDay, 1, "Checking first of the month date");
+    assert.equal(this.testmonth.firstDayIndex, 3, "Checking index of the first of the month");
+    
+    assert.deepEqual(this.testmonth.dayIndex, {}, "Initialized the day Index object");
+    assert.equal(this.testmonth.monthId, "20171", "Initialized the month Id");
+});
+
+
+// Testing Calendar Helper functions
+QUnit.module( "Calendar helper functions tests", {
+});
+
+QUnit.test("generateUniqueId test", function(assert) {
+    assert.expect(1);
+    var uniqueId = generateUniqueId();
+    var uniqueId2 = generateUniqueId();
+    
+    assert.notEqual(uniqueId, uniqueId2);
+});
+
+QUnit.test("emptyCalendarState test", function(assert) {
+    assert.expect(6);
+    var params = {startDate: "2017-02-14" , endDate: "2017-02-20" , calendarTitle: ""};
+    var state = emptyCalendarState(params);
+    
+    assert.equal(state.title, "", "Checking state title");
+    assert.equal(state.startDateString, "20170214", "Checking startDateString");
+    assert.equal(state.endDateString, "20170220", "Checking endDateString");
+    // What kind of assertion for uniqueId?
+    assert.deepEqual(state.checkedDays, {});
+    
+    //assert that an exception is thrown with invalid dates.
+    var params = {startDate: "2017-02-14", endDate: "2017-01-01", calendarTitle: "hello"};
+    // I don't understand the qunit api for assert.raises (or assert.throws)
```

Comment:
How does it work? Although, I followed along with the example they had when I wrote this.